File: frontend/lib/core/models/teacher.dart
class Teacher {
  final int id;
  final String fullName;
  final int subjectId;
  final String avatarUrl;
  final String bio;

  const Teacher({
    required this.id,
    required this.fullName,
    required this.subjectId,
    required this.avatarUrl,
    required this.bio,
  });

  factory Teacher.fromJson(Map<String, dynamic> json) {
    return Teacher(
      id: json['id'] as int,
      fullName: json['full_name'] as String,
      subjectId: json['subject_id'] as int? ?? 0,
      avatarUrl: json['avatar_url'] as String? ?? '',
      bio: json['bio'] as String? ?? '',
    );
  }
}

File: backend/internal/models/teacher.go
package models

import "time"

type Teacher struct {
	ID        uint      `json:"id"`
	FullName  string    `json:"full_name"`
	SubjectID uint      `json:"subject_id"`
	AvatarURL string    `json:"avatar_url"`
	Bio       string    `json:"bio"`
	CreatedAt time.Time `json:"created_at"`
}
